posthog/models/sessions/sql.py

```diff
@@ -0,0 +1,130 @@
+from django.conf import settings
+
+from posthog.clickhouse.table_engines import (
+    Distributed,
+    ReplicationScheme,
+    AggregatingMergeTree,
+)
+
+TABLE_BASE_NAME = "sessions"
+SESSIONS_DATA_TABLE = lambda: f"sharded_{TABLE_BASE_NAME}"
+
+# if updating these column definitions
+# you'll need to update the explicit column definitions in the materialized view creation statement below
+SESSIONS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    -- part of order by so will aggregate correctly
+    session_id VARCHAR,
+    -- part of order by so will aggregate correctly
+    team_id Int64,
+    -- ClickHouse will pick any value of distinct_id for the session
+    -- this is fine since even if the distinct_id changes during a session
+    -- it will still (or should still) map to the same person
+    distinct_id SimpleAggregateFunction(any, String),
```

Comment:
It does seem like we don't need this. I don't think there's a situation where we can only query the sessions table, I think we always need to join with the events table. Same for person_id

Reply:
I've been convinced that we should keep this around, there's little down-side and potentially in the future we'll be glad we had it, for e.g. backfilling a person property, or e.g. hogql could allow sessions.person.properties.some_property.

Reply:
10000%